Source/ARTLocalDevice.m

```diff
@@ -75,21 +75,20 @@ + (ARTLocalDevice *)load:(NSString *)clientId storage:(id<ARTDeviceStorage>)stor
     NSString *deviceId = [storage objectForKey:ARTDeviceIdKey];
     NSString *deviceSecret = deviceId == nil ? nil : [storage secretForDevice:deviceId];
     
-    if (deviceId == nil || deviceSecret == nil) { // generate both at the same time
-        deviceId = [self generateId];
-        deviceSecret = [self generateSecret];
-        
-        [storage setObject:deviceId forKey:ARTDeviceIdKey];
-        [storage setSecret:deviceSecret forDevice:deviceId];
-    }
-    
-    device.id = deviceId;
+    device.id = deviceId ?: [self.class generateId]; // temporarily ignore RSH8a/b, see https://github.com/ably/ably-cocoa/pull/1847#discussion_r1441954284 thread
     device.secret = deviceSecret;
 
     id identityTokenDetailsInfo = [storage objectForKey:ARTDeviceIdentityTokenKey];
     ARTDeviceIdentityTokenDetails *identityTokenDetails = [ARTDeviceIdentityTokenDetails unarchive:identityTokenDetailsInfo withLogger:logger];
     device->_identityTokenDetails = identityTokenDetails;
 
+    NSString *clientId = [storage objectForKey:ARTClientIdKey];
+    if (clientId == nil && identityTokenDetails.clientId != nil) {
```

Comment:
This is not a specified behaviour — please could you add a comment explaining why we do it?

Reply:
https://github.com/ably/ably-cocoa/pull/1847/commits/e9cab44214014d599353bb8b740fd1c1e5ba77c6